### PR TITLE
fix: set sentAt & separate timestamp and originalTimestamp

### PR DESCRIPTION
**Pull request by rhino1998**
# Description

When `timestamp` is not specified, `timestamp` is computed as `timestamp = receivedAt - (sentAt - originalTimestamp)`. Currently this is handled incorrectly as `sentAt` was set to the same value `originalTimestamp` causing the computed `timestamp` to be `receivedAt`. Manually setting `originalTimestamp` was not functional as it `sentAt` was still set to that value.

This PR makes the following changes
* Sets `sentAt` at batch-send time rather than at message-enqueue time so that the remote skew calculation works

### CODE REVIEW

File: message.go [hunk 1]
```diff
@@ -47,15 +47,6 @@ func makeMessageId(id string, def string) string {
 	return id
 }
 
-// Returns the time value passed as first argument, unless it's the zero-value,
-// in that case the default value passed as second argument is returned.
-func makeTimestamp(t time.Time, def time.Time) time.Time {
```
Comment (hermanschaaf) [inline message.go]: Why did you remove this function? Seems like it's not related to the underlying bug

File: analytics.go [hunk 2]
```diff
@@ -375,6 +369,16 @@ func (c *client) getMarshalled(msgs []message) ([]byte, error) {
 func (c *client) send(msgs []message, retryAttempt int) {
 	const attempts = 10
 
+	ts := c.now()
+	for i := range msgs {
+		err := msgs[i].setSentAt(ts, c.MaxMessageBytes)
+		if err != nil {
+			c.errorf("%s - %v", err, msgs[i].msg)
+			c.notifyFailure([]message{msgs[i]}, err)
+			break
```
Comment (hermanschaaf) [inline analytics.go]: rather than break here on error, should we not continue and still try to set `sentAt` on the other messages?

File: message.go [hunk 3]
```diff
@@ -83,6 +74,37 @@ func (m message) MarshalJSON() ([]byte, error) {
...
+	case Page:
+		msg.SentAt = ts
+		m.msg = msg
+	case Screen:
+		msg.SentAt = ts
+		m.msg = msg
+	case Track:
+		msg.SentAt = ts
+		m.msg = msg
+	}
```
Comment (hermanschaaf) [inline message.go:111]: it would be good to add a default case here that errors out with something like

```
	default:
		err = fmt.Errorf("messages with custom types cannot be enqueued: %T", msg)
		return
```

File: message.go (hunk 1 above)
Comment (rhino1998) [reply message.go]: It was left unused after the bug was fixed. It was just used to set OriginalTimestamp to a default if it wasn't already defined. Actually I should probably re-add that to allow submitting historical events but still with clock skew correction

File: message.go (hunk 3 above)
Comment (rhino1998) [reply message.go:111]: [d8cce36](https://github.com/cloudquery/analytics-go/pull/1/commits/d8cce367dbbbb674f1f25257b838c988388dab31)

File: message.go (hunk 1 above)
Comment (hermanschaaf) [reply message.go]: Right, yeah I think let's keep the `m.OriginalTimestamp = makeTimestamp(m.OriginalTimestamp, ts)` lines, so we're only changing what we need to

File: analytics.go (hunk 2 above)
Comment (rhino1998) [reply analytics.go]: [3cccb00](https://github.com/cloudquery/analytics-go/pull/1/commits/3cccb00027231f4976b438380ba901c16612b316)

File: message.go (hunk 1 above)
Comment (rhino1998) [reply message.go]: [404997c](https://github.com/cloudquery/analytics-go/pull/1/commits/404997ca0c26840c82bc92bc9718e453c5fa59cc)

**Review by hermanschaaf — APPROVED** [review]
Nice 🚀
